### ISSUE TITLE
JSStackTrace gets lineNumber without judging the empty state, which will cause sometimes JavaScriptException error printing information is NoSuchKeyException

### ISSUE THREAD

**Issue by Nichooool** [facebook/react-native#24539]
## 🐛 Bug Report
JSStackTrace gets lineNumber without judging the empty state, which will cause sometimes JavaScriptException error printing information is NoSuchKeyException

## To Reproduce
![image](https://user-images.githubusercontent.com/26821180/56455832-6eff9a80-6396-11e9-8533-78f51f080d61.png)
The Release package will output NoSuchKeyException lineNumber does not exist.

## Expected Behavior
Release package output JavascriptException

## Repair Plan
I can fix the problem by modifying the JSStackkTrace by downloading the source code.
![image](https://user-images.githubusercontent.com/26821180/56455808-07e1e600-6396-11e9-9761-0ba8b3df0f7b.png)

```
04-20 17:59:39.484 19069-19157/com.xxx.xxx W/cmcm: error >> com.facebook.react.common.JavascriptException: Can't find variable: csafkljdal, stack:
    onPress@500:826
    touchableHandlePress@204:1584
    touchableHandlePress@unknow:unknow
    _performSideEffectsForTransition@196:8945
    _performSideEffectsForTransition@unknow:unknow
    _receiveSignal@196:7658
    _receiveSignal@unknow:unknow
    touchableHandleResponderRelease@196:5046
    touchableHandleResponderRelease@unknow:unknow
    y@90:576
    P@90:719
    E@90:773
    M@90:1940
    H@90:2699
    forEach@unknow:unknow
    j@90:2499
    <unknown>@90:14003
    Ue@90:83341
    De@90:13673
    We@90:13846
    receiveTouches@90:14605
    value@28:3311
    <unknown>@28:822
    value@28:2565
    value@28:794
    value@unknow:unknow
    
        at com.facebook.react.modules.core.ExceptionsManagerModule.showOrThrowError(ExceptionsManagerModule.java:54)
        at com.facebook.react.modules.core.ExceptionsManagerModule.reportFatalException(ExceptionsManagerModule.java:38)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
        at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:158)
        at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
        at android.os.Handler.handleCallback(Handler.java:751)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:29)
        at android.os.Looper.loop(Looper.java:154)
        at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:232)
        at java.lang.Thread.run(Thread.java:761)
```
## Environment
ReactNative v0.59.3
Android  7.1.2

**Comment by JKCooper2** [comment]
Doesn't it currently just put -1 instead of 'unknown'? https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/util/JSStackTrace.java#L36

**Comment by wbercx** [comment]
This was resolved in [0.59.5](https://github.com/facebook/react-native/releases/tag/v0.59.5).

**Comment by Nichooool** [comment]
> This was resolved in [0.59.5](https://github.com/facebook/react-native/releases/tag/v0.59.5).

Well, thank you for your reminder, I only saw the 59.5 fix, we used 59.3, I am upgrading 59.5 now.

**Comment by Nichooool** [comment]
> Doesn't it currently just put -1 instead of 'unknown'? https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/util/JSStackTrace.java#L36



> Doesn't it currently just put -1 instead of 'unknown'? https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/util/JSStackTrace.java#L36

Thank you
